### ISSUE TITLE
floating point model for Alexnet

### ISSUE THREAD

**Issue by AhmedTElthakeb** [jiecaoyu/XNOR-Net-PyTorch#71]
How can I run floating point model on alexnet?
Is there a way to disable the binarization (for the sake of testing) and just run the pre-trained model in full precision?
I want to get these fp accuracies with this implementation (arch):
Top-1: 57.1% Top-5: 80.2%

**Comment by jiecaoyu** [comment]
Hi @AhmedTE ,

(1) remove weight binarization [```bin_op.binarization()```](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/ebccb503b2f9ebc96187a75b6daa63d95b61d110/ImageNet/networks/main.py#L244)

(2) remove activation binarization [```x = BinActive()(x)```](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/ebccb503b2f9ebc96187a75b6daa63d95b61d110/ImageNet/networks/model_list/alexnet.py#L51)

Then you need to retrain the network from scratch. There is no direct way to get the pre-trained model since XNOR-Net changes the order of CONV and BN.

**Comment by AhmedTElthakeb** [comment]
Hi jiecaoyu,

Thanks very much for your quick response!

Do you by chance have such pre-trained fp model that you may share with me?
That would be extremely helpful, otherwise thanks again.

**Comment by AhmedTElthakeb** [comment]
I actually commented out (1) and (2) as you mentioned, and trained lenet from scratch, here is what I get:
Test set: Average loss: 0.7732, Accuracy: 7807/10000 (78.07%)
Best Accuracy: 93.83%

Is it the case that such network topology (changing the order of CONV and BN) is particularly optimized for binary weights only? have you experienced similar behavior?

**Comment by jiecaoyu** [comment]
@AhmedTE I don't have the pretrained AlexNet. The network topology is optimized for binary networks.

However, if you also remove the [line 44 & 45](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/ebccb503b2f9ebc96187a75b6daa63d95b61d110/MNIST/main.py#L44). You can run the training as

```bash
python main.py --lr 0.001 --wd 3e-5
```

It gives an accuracy of:
```
Test set: Average loss: 0.0529, Accuracy: 9915/10000 (99.15%)
Best Accuracy: 99.18%
```

**Comment by AhmedTElthakeb** [comment]
Hi @jiecaoyu,
Thanks very much for your quick responses and help. That works perfect.
